test/system/trader.js

```diff
@@ -113,34 +160,65 @@ class Trader {
       if (side === 'B') {
         let calcCoins = this.calcMaxCoinsToBuy(price);
         if (calcCoins.coinsToBuy === 0) {
-          console.log('Not enough capital to initiate order');
+          console.log('Not enough capital to initiate order', this.config.name);
           resolve(true);
         }
         else {
           this.updateBalance('ASSETS', currentAssets + calcCoins.coinsToBuy);
           this.updateBalance('CAPITAL', currentCapital - calcCoins.totalCost);
+          this.getLastBuyPriceAndQuntity(price, calcCoins.coinsToBuy);
           resolve(this.createOrder(price, side, calcCoins.coinsToBuy));
         }
       }
       else {
-        const maxAssetsToUse = currentAssets * (this.config.balance.maxAssetsToUse / 100);
-        if (currentAssets > 0) {
-          let assetsToSell = maxAssetsToUse;
-          let totalCost = assetsToSell * price;
-          this.updateBalance('ASSETS', currentAssets - assetsToSell);
-          this.updateBalance('CAPITAL', currentCapital + totalCost);
-          resolve(this.createOrder(price, side, assetsToSell));
+        const maxAssetsToUse = currentCapital * (this.config.balance.maxAssetsToUse / 100);
+        if (maxAssetsToUse > 0) {
+          if(this.config.name === 'trader2') {
+            let quntities = orderbook.quantities.map(q => {
+              return q;
+            });
+
+            if(quntities.length){
+              const assetsToSell =  Math.max(...quntities);
+              if(assetsToSell < currentCapital) {
+                let totalCost = assetsToSell * price;
+                this.updateBalance('ASSETS', currentAssets - assetsToSell);
+                this.updateBalance('CAPITAL', currentCapital + totalCost);
+                resolve(this.createOrder(price, side, assetsToSell / 2));
+              }
+              else {
+                console.log('don\'t have enough assets to sell', this.config.name);
+                resolve(true);
+              }
+            }
+
+          } else {
+            let assetsToSell = maxAssetsToUse;
+            let totalCost = assetsToSell * price;
+            this.updateBalance('ASSETS', currentAssets - assetsToSell);
+            this.updateBalance('CAPITAL', currentCapital + totalCost);
+            resolve(this.createOrder(price, side, assetsToSell));
+          }
         }
         else {
-          console.log('don\'t have an assets to sell');
+          console.log('don\'t have an assets to sell', this.config.name);
```

Comment:
Tip - a nicer way to have apostrophes in strings is to use double quotes:
"I don't have any assets to sell"